src/actions/tournaments.ts

```diff
@@ -1 +1,140 @@
-import { RootState } from '../reducers';
+import { RootState } from '../store';
+import { API_TOURNAMENTS_URL } from '../constants/api';
+import { Tournament } from '../reducers/tournaments';
+import { ThunkAction } from 'redux-thunk';
+import { AnyAction } from 'redux';
+
+type Thunk = ThunkAction<void, RootState, unknown, AnyAction>;
+
+const fetchTournamentsStarted = () => ({
+  type: 'tournaments/fetchStarted' as const,
+});
+
+const fetchTournamentsSuccess = (tournaments: Tournament[]) => ({
+  type: 'tournaments/fetchSucceeded' as const,
+  payload: tournaments,
+});
+
+const fetchTournamentsFailed = (error: unknown) => ({
+  type: 'tournaments/fetchFailed' as const,
+  error,
+});
+
+const fetchTournaments = (): Thunk => async (dispatch, getState) => {
+  dispatch(fetchTournamentsStarted());
+
+  // The search here may return unexpected tournaments because it searches _all_ fields.
+  // This includes the ISO start date and id fields at the time of writing.
+  // It appears that the backend doesn't support searching specific fields. (filters are exact matches only)
+
+  try {
+    const base = API_TOURNAMENTS_URL;
+    const search = getState().tournaments.search;
+    const url = search ? `${base}?q=${search}` : base;
+    const response = await fetch(url);
```

Comment:
I'd generally expect to be using `axios` (or a similar http request library) for browser compatibility and ease of use.
The code here would barely change though, so I figured this would work well enough for the challenge.